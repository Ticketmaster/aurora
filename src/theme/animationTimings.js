const listTimings = {
  caretRotate: 100,
  collapse: 300,
  contentFadeOut: 100,
  contentFadeIn: 300,
  contentFadeInDelay: 100
};

export default listTimings;
