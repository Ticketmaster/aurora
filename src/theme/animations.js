import { css } from "styled-components";
import CONSTANTS from "./constants";

const { easing: EASING } = CONSTANTS;

const borderRadius = css`margin-bottom 0.3s ${EASING.easeInOutQuad} 0s`
const boxShadow = css`box-shadow 0.3s ${EASING.easeInOutQuad} 0s`;
const marginBottom = css`margin-bottom 0.3s ${EASING.easeInOutQuad} 0s`;
const maxHeight = css`max-height 0.3s ${EASING.easeInOutQuad} 0s`;
const opacity = css`opacity 0.3s cubic-bezier(0.455, 0.03, 0.515, 0.955) 0.2s`;

const accordionSectionText = css`
  .collapse {
    opacity: 0;
    transition: opacity 0.1s cubic-bezier(0.55, 0.085, 0.68, 0.53) 0s;
    visibility: hidden;
  }
  .expand {
    opacity: 1;
    transition: opacity 0.3s cubic-bezier(0.455, 0.03, 0.515, 0.955) 0.2s;
    visibility: visible;
  }
`;

const minTransitionHeight = css`
  height: auto;
  max-height: 300px;
  transition: ${maxHeight};
`;

const maxTransitionHeight = css`
  max-height: 600px;
  transition: ${maxHeight};
`

const adjustHeight = css`
   .minHeight { ${minTransitionHeight}}
   .maxHeight { ${maxTransitionHeight} }
`;

const collapse = css`
  max-height: 0;
  opacity: 0;
  visibility: collapse;
  transition: ${maxHeight}, ${opacity}, ${marginBottom};
`;
const expand = css`
  max-height: 600px;
  opacity: 1;
  visibility: visible;
  transition: ${maxHeight}, ${opacity}, ${marginBottom};
`;

const expandCollapse = css`
  .expand { ${expand} }
  .collapse { ${collapse} }
`;

const hover = css`
  background-color: rgba(2, 108, 223, 0.1);
  -webkit-transition: background-color 100ms linear;
  -ms-transition: background-color 100ms linear;
  transition: background-color 100ms linear;
`;

export {
  adjustHeight,
  accordionSectionText,
  borderRadius,
  boxShadow,
  collapse,
  expand,
  expandCollapse,
  hover,
  marginBottom,
  maxHeight,
  opacity
};
