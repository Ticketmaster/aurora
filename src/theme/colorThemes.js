const success = "#1bab1e";

const global = {
  brand: "#026cdf",
  primary: {
    // brand colors
    base: "#026cdf",
    reverse: "#3396ff",
    reverseLight: "#80b5ef",
    reverseLightest: "#bfdaf6",

    // button colors
    dark: "#0150a7",
    darkest: "#013670",
    light: "#d6e7fa",
    lightest: "#ebf4fd",

    // not used in new components. kept for backward compatibility.
    lightBase: "rgba(2, 108, 223, 0.1)",
    medium: "#0150a7",
    muted: "#d6e7fa"
  },
  special: {
    base: success,
    dark: "#23702c",
    darkest: "#265333",

    medium: "#23702c" // not used in new components. kept for backward compatibility.
  },
  darkFill: "#1f262d", // not used in new components. kept for backward compatibility.
  dark: {
    base: "#1f262d",
    light: "#8f9296",
    lightest: "#e3e4e5"
  },
  accent01: {
    base: "#c56bff",
    dark: "#904eba",
    light: "#eacbff",
    lightest: "#f9f0ff"
  },
  accent02: {
    base: "#962d94",
    dark: "#6d236c",
    light: "#dfc0df",
    lightest: "#f4eaf4"
  },
  accent03: {
    base: "#14a1a3",
    dark: "#0a7e80",
    light: "#aadddd",
    lightest: "#cef2f2",
    reverse: "#20c4c7"
  },
  accent04: "#00ffff",
  gray01: "#262626",
  gray02: "#999999",
  gray03: "#bfbfbf",
  gray04: "#ebebeb",
  gray05: "#f6f6f6",
  error: {
    base: "#d93a3a",
    dark: "#a22b2b",
    light: "#f3c3c3",
    lightest: "#fbebeb"
  },
  caution: {
    base: "#f2bd2a",
    dark: "#c69a22",
    light: "#fae7b4",
    lightest: "#fdf5df"
  },
  success: {
    base: success,
    dark: "#148016",
    light: "#bae5bb",
    lightest: "#e8f6e8"
  },
  transparent: "transparent",
  onyx: {
    base: "rgba(38, 38, 38, 1)",
    light: "rgba(38, 38, 38, 0.65)",
    muted: "rgba(38, 38, 38, 0.4)"
  },
  white: {
    base: "rgba(255, 255, 255, 1)",
    lighter: "rgba(255, 255, 255, 0.2)",
    light: "rgba(255, 255, 255, 0.7)",
    muted: "rgba(255, 255, 255, 0.5)"
  }
};

const themes = {
  global,
  tm: global,
  lne: {
    ...global,
    brand: "#e11d39",
    primary: {
      base: "#2cb3ad",
      medium: "#239893",
      dark: "#1f8783",
      light: "#e9f7f6",
      lightest: "#e9f7f6",
      muted: "#d5f0ef"
    }
  }
};

export default themes;
