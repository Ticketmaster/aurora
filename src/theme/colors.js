import themes from "./colorThemes";

const { tm } = themes;

const colors = {
  // Deprecated colors. Should be removed
  azure: {
    dark: tm.primary.base,
    base: tm.primary.base,
    light: "rgba(2, 108, 223, 0.2)",
    muted: tm.primary.lightBase,
    restrained: "rgba(2, 108, 223, 0.4)",
    hover: tm.primary.medium,
    hoverLight: tm.primary.light,
    pressed: tm.primary.dark,
    pressedLight: tm.primary.muted
  },
  emerald: {
    ...tm.success,
    hover: tm.special.medium,
    pressed: tm.special.dark,
    muted: "rgba(27, 171, 30, 0.4)",
    restrained: "rgba(27, 171, 30, 0.2)"
  },
  outlined: {
    hover: "#ECF4FD",
    pressed: "#D9E8F9"
  },
  transparent: tm.transparent,
  summerSky: "#3AC7FF",
  blackPearl: tm.darkFill,
  slate: tm.gray02,
  moonstone: "#D1D1D1",
  shale: "#E6E6E6",
  quartz: tm.gray05,
  lightGray: "#E0E0E0",
  aquamarine: tm.accent04,
  turquoise: tm.accent03,
  ruby: {
    base: tm.error.base,
    darkRuby: tm.error.dark,
    lightRuby: tm.error.light
  },
  amber: {
    base: tm.caution.base,
    darkAmber: tm.caution.dark,
    lightAmber: tm.caution.light
  },
  moonrock: tm.gray03,
  diatomite: tm.gray04,
  heliotrope: {
    ...tm.accent01,
    muted: "#F3E1FF"
  },
  cruz: {
    ...tm.accent02,
    muted: "#e3cae3"
  },
  alert: {
    base: tm.error.base,
    dark: "#AE393A",
    light: "#EC9C9C",
    muted: "#F5CDCD"
  },
  caution: tm.caution,
  positive: {
    base: tm.success.base,
    dark: "#208E25",
    light: "#8DD58E",
    muted: "#C5E9C6"
  },
  defaultGradient: {
    from: tm.primary.base,
    to: "#3ac7ff"
  },
  // Onyx and white are specific for typography. Should not be removed
  onyx: tm.onyx,
  white: tm.white
};

export default colors;
