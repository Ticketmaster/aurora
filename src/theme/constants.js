const constants = {
  borderRadius: {
    small: "2px",
    large: "4px"
  },
  breakpoint: {
    small: "480px",
    medium: "768px",
    large: "1024px"
  },
  pillBorderRadius: "18px",
  thumbnailSize: "40px",
  drawer: {
    dismissAreaHeight: "60px",
    lineUpAreaHeight: "55px",
    desktopWidth: 26.05,
    mobileWidth: 80
  },
  device: {
    mobile: `(min-width: 480px)`,
    tablet: `(min-width: 768px)`,
    desktop: `(min-width: 1024px)`
  }
};

export default constants;
