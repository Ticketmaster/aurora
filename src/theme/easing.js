export const bezierEasy = "cubic-bezier(0.455, 0.03, 0.515, 0.955)";
export const bezierExit = "cubic-bezier(0.55, 0.085, 0.68, 0.53)";
