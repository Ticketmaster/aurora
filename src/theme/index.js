export { default as animations } from "./animations";
export { default as colors } from "./colors";
export { default as constants } from "./constants";
export { default as spacing } from "./spacing";
export { default as themes } from "./colorThemes";
export { default as typography } from "./typography";
export { default as zIndex } from "./zIndex";
