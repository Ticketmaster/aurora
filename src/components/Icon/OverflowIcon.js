import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { constants } from "../../theme";
import Icon from "./";

const OverflowIconContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  @media only screen and ${constants.device.mobile} {
    display: none;
  }
`;

const OverflowIcon = ({ name, size, color, svgStyle }) => (
  <OverflowIconContainer>
    <Icon name={name} size={size} color={color} svgStyle={svgStyle} />
  </OverflowIconContainer>
);

OverflowIcon.defaultProps = {
  svgStyle: {}
};

OverflowIcon.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired,
  svgStyle: PropTypes.objectOf(PropTypes.any)
};

export default OverflowIcon;
