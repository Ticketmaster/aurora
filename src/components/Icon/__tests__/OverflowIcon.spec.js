import React from "react";
import renderer from "react-test-renderer";

import { colors } from "../../../theme";
import OverflowIcon from "../OverflowIcon";

describe("<OverflowIcon />", () => {
  it("renders OverflowIcon correctly", () => {
    const component = renderer.create(
      <OverflowIcon
        name="dots"
        size={15}
        color={colors.cascade.light}
        desktopDisplay={false}
        mobileDisplay
        svgStyle={{ transform: "rotate(-90deg) " }}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
