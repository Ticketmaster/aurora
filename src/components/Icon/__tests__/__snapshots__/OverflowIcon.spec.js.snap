// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OverflowIcon /> renders OverflowIcon correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media only screen and (min-width:480px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
>
  <svg
    height="15px"
    style={
      Object {
        "transform": "rotate(-90deg) ",
      }
    }
    viewBox="0 0 1024 1024"
    width="15px"
  >
    <path
      d="M82.33 635.904c-16.333-7.168-30.618-16.845-42.906-28.928s-21.914-26.317-28.928-42.598c-7.014-16.333-10.496-33.792-10.496-52.378s3.482-36.147 10.547-52.634c7.014-16.486 16.64-30.771 28.928-42.854s26.573-21.658 42.906-28.672c16.333-7.014 33.741-10.547 52.378-10.547s36.045 3.482 52.378 10.547c16.333 7.014 30.618 16.589 42.906 28.672s21.914 26.368 28.928 42.906c6.963 16.435 10.445 33.997 10.445 52.582s-3.482 36.045-10.547 52.378c-7.014 16.333-16.64 30.515-28.928 42.598s-26.573 21.76-42.906 28.928c-16.333 7.168-33.741 10.803-52.378 10.803s-35.994-3.584-52.326-10.803zM459.622 635.904c-16.333-7.168-30.618-16.845-42.906-28.928s-21.914-26.317-28.928-42.598c-7.014-16.333-10.496-33.792-10.496-52.378s3.482-36.147 10.547-52.634c7.014-16.486 16.64-30.771 28.928-42.854s26.573-21.658 42.906-28.672c16.282-7.066 33.741-10.547 52.326-10.547s36.045 3.482 52.378 10.547c16.333 7.014 30.618 16.589 42.906 28.672s21.914 26.368 28.928 42.906c7.014 16.486 10.547 33.997 10.547 52.634s-3.482 36.045-10.547 52.378c-7.014 16.333-16.64 30.515-28.928 42.598s-26.573 21.76-42.906 28.928c-16.333 7.168-33.741 10.803-52.378 10.803s-36.045-3.635-52.378-10.854zM836.659 635.904c-16.486-7.168-30.771-16.845-42.906-28.928s-21.658-26.317-28.672-42.598c-7.014-16.333-10.547-33.741-10.547-52.378s3.482-36.147 10.547-52.634c7.014-16.486 16.589-30.771 28.672-42.906 12.083-12.083 26.368-21.658 42.906-28.672 16.486-7.014 33.997-10.547 52.634-10.547 18.586 0 36.045 3.482 52.378 10.547 16.333 7.014 30.515 16.589 42.598 28.672s21.76 26.368 28.928 42.906c7.219 16.486 10.803 34.048 10.803 52.634s-3.584 36.045-10.803 52.378c-7.168 16.333-16.845 30.515-28.928 42.598s-26.317 21.76-42.598 28.928c-16.333 7.168-33.741 10.803-52.378 10.803-18.586 0-36.147-3.584-52.634-10.803z"
      fill="#b4b9bd"
    />
  </svg>
</div>
`;
